x/agent.py

```diff
@@ -38,14 +38,21 @@ class DiscreteAgent(Agent):
 
     """
     def __init__(self, model, memory, epsilon=None):
-        super(DiscretAgent, self).__init__(model, memory)
+        super(DiscreteAgent, self).__init__(model, memory)
         if epsilon is None:
             self.epsilon = lambda *args: .1
 
-    def compile(self, optimizer="sgd", loss="mse", policy_rule="max",
-                experience=None):
-        self.model.compile(optimizer, loss, policy_rule)
+    def compile(self, *args, **kwargs):
+        self.model.compile(*args, **kwargs)
+        if 'experience' in kwargs:
+            experience = kwargs['experience']
+        else:
+            experience = None
         self.memory.reset(experience)
+#    def compile(self, optimizer="sgd", loss="mse", policy_rule="max",
+#                experience=None):
+#        self.model.compile(optimizer, loss, policy_rule)
+#        self.memory.reset(experience)
```

Comment:
Why did you remove this? Also, I think it is better if we don't push commented out code.